client/orm/filter/opentelemetry/filter.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 beego
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentelemetry
+
+import (
+	"context"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	otelTrace "go.opentelemetry.io/otel/trace"
+	"strings"
+
+	"github.com/beego/beego/v2/client/orm"
+)
+
+type CustomSpanFunc func(span otelTrace.Span, ctx context.Context, inv *orm.Invocation)
+
+// FilterChainBuilder provides an opentelemtry Filter
+type FilterChainBuilder struct {
+	// customSpanFunc users are able to custom their span
+	customSpanFunc CustomSpanFunc
+}
+
+func NewFilterChainBuilder() *FilterChainBuilder {
+	return &FilterChainBuilder{}
+}
+
+func (builder *FilterChainBuilder) SetCustomSpanFunc(spanFunc CustomSpanFunc) *FilterChainBuilder {
+	builder.customSpanFunc = spanFunc
+	return builder
+}
```

Comment:
remove this function and change to use option pattern. You can learn the Go option pattern via google